packages/SwingSet/src/kernel/slogger.js

```diff
@@ -0,0 +1,111 @@
+/* global harden */
+import { assert } from '@agoric/assert';
+
+const IDLE = 'idle';
+const STARTUP = 'startup';
+const DELIVERY = 'delivery';
+
+export function makeDummySlogger(makeConsole) {
+  return harden({
+    addVat: () => 0,
+    vatConsole: () => makeConsole('disabled slogger'),
+    startup: () => () => 0, // returns nop finish() function
+    delivery: () => () => 0,
+    syscall: () => () => 0,
+  });
+}
+
+export function makeSlogger(writeObj) {
+  const write = writeObj ? e => writeObj(harden(e)) : () => 0;
+
+  const vatSlogs = new Map(); // vatID -> vatSlog
+
+  function makeVatSlog(vatID) {
+    let state = IDLE; // or STARTUP or DELIVERY
+    let crankNum;
+    let deliveryNum = 0;
+    let syscallNum;
+
+    function assertOldState(exp, msg) {
+      assert(state === exp, `vat ${vatID} in ${state}, not ${exp}: ${msg}`);
+    }
+
+    function vatConsole(origConsole) {
+      const vc = {};
+      for (const level of ['debug', 'log', 'info', 'warn', 'error']) {
```

Comment:
I think I fundamentally don't buy in to the traditional hierarchy of logging levels, but this is a longer conversation to have before we land this.

Reply:
Yeah, the goal here is simply to match the functionality of the original `console` object. This code is actually copied from the `anylogger` wrapper that @michaelfig put together, which we might want to augment with the more exotic `console` methods (there are some timestamping ones that we omitted).